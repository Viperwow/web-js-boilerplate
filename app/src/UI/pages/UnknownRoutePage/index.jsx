// Vendors
import React, {Component} from 'react';
// Constants
const UNKNOWN_PAGE_PLACEHOLDER = 'UNKNOWN_PAGE_PLACEHOLDER';

class UnknownRoutePage extends Component {
  render() {
    return (
      <div className="page">
        {UNKNOWN_PAGE_PLACEHOLDER}
      </div>
    );
  }
}

export default UnknownRoutePage;
