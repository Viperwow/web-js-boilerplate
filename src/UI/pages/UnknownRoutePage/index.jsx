// Vendors
import React, {Component} from 'react';
import i18n from 'i18next';

class UnknownRoutePage extends Component {
  render() {
    return (
      <div className="page">
        {i18n.t('pages.unknown')}
      </div>
    );
  }
}

export default UnknownRoutePage;
