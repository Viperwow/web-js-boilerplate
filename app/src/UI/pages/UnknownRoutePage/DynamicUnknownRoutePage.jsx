// HoC
import loadable from 'src/common/loadable';

const DynamicUnknownRoutePage = loadable(
  import('src/UI/pages/UnknownRoutePage' /* webpackChunkName: "UnknownRoutePage" */),
);

export default DynamicUnknownRoutePage;
