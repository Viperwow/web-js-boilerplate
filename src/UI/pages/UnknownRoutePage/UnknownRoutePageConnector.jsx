// Vendors
import i18n from 'i18next';
// Common
import queried from 'src/common/queried';
// GQL
import {QUERY_LOCALE} from 'src/client-gql/locale';
// UI
import UnknownRoutePage from '.';

const mapStateToProps = () => ({
  text: i18n.t('pages.unknown'),
});

export default queried(QUERY_LOCALE, mapStateToProps)(UnknownRoutePage);
