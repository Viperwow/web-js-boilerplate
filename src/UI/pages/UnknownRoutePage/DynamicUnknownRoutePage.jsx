// Vendors
import React from 'react';
// HoC
import loadable from 'src/common/loadable';

const DynamicUnknownRoutePage = props => {
  const LoadedUnknownRoutePage = loadable(
    import('src/UI/pages/UnknownRoutePage/UnknownRoutePageConnector' /* webpackChunkName: "UnknownRoutePage" */), // eslint-disable-line max-len
  );

  return <LoadedUnknownRoutePage {...props} />;
};

export default DynamicUnknownRoutePage;
