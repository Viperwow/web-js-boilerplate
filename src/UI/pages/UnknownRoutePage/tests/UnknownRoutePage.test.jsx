// Vendors
import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
// UI
import UnknownRoutePage from '..';

jest.unmock('..');

describe('UnknownRoutePage component should render correctly', () => {
  test('With default props', () => {
    const output = shallow(<UnknownRoutePage />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });

  test('With text prop', () => {
    const output = shallow(<UnknownRoutePage text="text" />);

    expect(shallowToJson(output))
      .toMatchSnapshot();
  });
});
