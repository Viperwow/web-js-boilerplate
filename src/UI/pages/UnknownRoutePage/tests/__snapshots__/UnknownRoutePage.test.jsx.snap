// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnknownRoutePage component should render correctly With default props 1`] = `
<div
  className="page"
/>
`;

exports[`UnknownRoutePage component should render correctly With text prop 1`] = `
<div
  className="page"
>
  text
</div>
`;
